app/controllers/security_group_controller.rb

```diff
@@ -14,4 +14,247 @@ def self.display_methods
   end
 
   menu_section :net
+
+  def button
+    @edit = session[:edit] # Restore @edit for adv search box
+    params[:display] = @display if %w(vms instances images).include?(@display)
+    params[:page] = @current_page unless @current_page.nil? # Save current page for list refresh
+
+    @refresh_div = "main_div"
+    return tag("SecurityGroup") if params[:pressed] == "security_group_tag"
+    delete_security_groups if params[:pressed] == 'security_group_delete'
+
+    if params[:pressed] == "security_group_edit"
+      checked_security_group_id = get_checked_security_group_id(params)
+      javascript_redirect :action => "edit", :id => checked_security_group_id
+    elsif params[:pressed] == "security_group_new"
+      javascript_redirect :action => "new"
+    elsif !flash_errors? && @refresh_div == "main_div" && @lastaction == "show_list"
+      replace_gtl_main_div
+    else
+      render_flash
+    end
+  end
```

Comment:
@martinpovolny, 

I understand the idea, meanwhile that's not fair, this is not introducing anything new code at all and certainly not making it worse either, just using plain old approach like 99% of the current existing `def button` code. I'm happy to help with the refactoring but into another PR.